torch/_inductor/autotuner/model.py

```diff
@@ -0,0 +1,649 @@
+    "isnan": 55,
+}
+
+
+class KernelCategory(enumerate):
+    POINTWISE = 0
+    REDUCTION = 1
+    PERSISTENT_REDUCTION = 2
+
+
+def get_kernel_category(src: str) -> KernelCategory:
+    if "@pointwise" in src:
+        return KernelCategory.POINTWISE
+    if "@reduction" in src:
+        return KernelCategory.REDUCTION
+    if "@persistent_reduction" in src:
+        return KernelCategory.PERSISTENT_REDUCTION
+
+
+def get_number_of_loops(src: str) -> int:
+    return src.count("for roffset in range(0, rnumel, RBLOCK):")
+
+
+def parse_list_of_numbers(s: str) -> list:
+    # num1, num2, num3, ...
+    nums = s.strip().split(",")
+    nums = [num.strip() for num in nums]
+    return [int(num) for num in nums]
+
+
+def get_size_hints(src: str) -> list:
+    return parse_list_of_numbers(re.search(r"size_hints=\[([^\]]*)\]", src).group(1))
+
+
+def get_tiling(src: str) -> list:
+    names = ["xnumel", "ynumel", "rnumel"]
+    result = list()
+    for name in names:
+        startpos = src.find(name + " =")
+        if startpos == -1:
+            result.append(1)
+            continue
+        endpos = src.find("\n", startpos)
+        result.append(int(src[startpos + len(name + " = ") : endpos]))
+    return result
+
+
+def pad_tensor():
+    tensor_feature = list()
+    tensor_feature.append(False)  # StarDepOrWeakDep
+    tensor_feature.append(0)  # bytes
+    # we use the lowest 6 dims of the tensor
+    tensor_feature.extend([0] * 6)  # strides,
+    # we use the lowest 6 dims of the tensor
+    tensor_feature.extend([0] * 6)  # size
+    tensor_feature.append(True)  # is_contiguous
+    tensor_feature.append(False)  # is_scalar
+    tensor_feature.append(False)  # is_indirect
+    return tensor_feature
+
+
+def tensor_list(deps, total_bytes, rw_len):
+    rw_list = list(
+        [
+            (dep, bytes)
+            for dep, bytes in zip(deps, total_bytes)
+            if not isinstance(dep, (StarDep, WeakDep))
+        ]
+    )
+    res = list()
+    # sort the tensors by bytes in descending order
+    rw_list = sorted(rw_list, key=lambda x: x[1], reverse=True)
+    # for dep, strides, sizes, bytes in rw_list[:rw_len]:
+    for dep, bytes in rw_list[:rw_len]:
+        tensor_feature = pad_tensor()
+        tensor_feature[0] = isinstance(dep, (StarDep, WeakDep))
+        tensor_feature[1] = bytes
+        # left pad strides, strides can be None if StarDep or WeakDep
+        # if strides is not None:
+        strides = V.graph.sizevars.stride_hints(dep.index, dep.var_names)
+        for i in range(len(strides)):
+            # we use the lowest 6 dims of the tensor
+            tensor_feature[8 - (len(strides) - i)] = strides[i]
+        # left pad size, sizes can be None if StarDep or WeakDep
+        # if sizes is not None:
+        sizes = [int(size_) for size_ in dep.size]
+        for i in range(len(sizes)):
+            # we use the lowest 6 dims of the tensor
+            tensor_feature[14 - (len(sizes) - i)] = sizes[i]
```

Comment:
This encoding will make broadcasting/indirect indexing hard to detect for the model.  A zero stride used for broadcasting is pretty different from a non-existent stride, but they both result in the same feature.